File: frontend/src/services/categoryService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/categories';

export const getAllCategories = () => axios.get(API_URL);
export const getCategoryById = (id) => axios.get(`${API_URL}/${id}`);
export const addCategory = (category) => axios.post(API_URL, category);
export const updateCategory = (id, category) => axios.put(`${API_URL}/${id}`, category);
export const deleteCategory = (id) => axios.delete(`${API_URL}/${id}`);
export const searchByName = (name) => axios.get(`${API_URL}/search?name=${name}`);

File: backend/src/main/java/com/hrplatform/service/CategoryService.java
package com.hrplatform.service;

import java.util.List;

import com.hrplatform.model.Category;

public interface CategoryService {

    Category addCategory(Category category);

    Category updateCategory(Long id, Category category);

    void deleteCategory(Long id);

    List<Category> getAllCategories();

    Category getCategoryById(Long id);

    List<Category> searchByName(String name);

}